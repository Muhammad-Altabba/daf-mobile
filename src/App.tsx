import React from 'react'
import { ApolloProvider } from 'react-apollo'
import codePush, { DownloadProgress } from 'react-native-code-push'
import { handleCodePushStatusChange } from './lib/CodepushHelpers'
import Navigation from './navigators'
import { client } from './lib/GraphQL'
import Log from './lib/Log'
import './lib/I18n'

import { ThemeProvider } from '@kancha/kancha-ui'
import { Theme } from './theme'

class App extends React.Component {
  codePushStatusDidChange(status: codePush.SyncStatus) {
    handleCodePushStatusChange(status)
  }

  codePushDownloadDidProgress(progress: DownloadProgress) {
    // console.log(progress.receivedBytes + ' of ' + progress.totalBytes + ' received.');
  }

  render() {
    return (
      <ApolloProvider client={client}>
        <ThemeProvider theme={Theme}>
          <Navigation />
        </ThemeProvider>
      </ApolloProvider>
    )
  }
}

export default App
