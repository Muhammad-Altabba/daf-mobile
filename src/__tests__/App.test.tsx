import 'react-native'
import React from 'react'
import App from '../App'
import { render } from 'react-native-testing-library'

jest.useFakeTimers()

it('renders correctly', () => {
  const tree = render(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
