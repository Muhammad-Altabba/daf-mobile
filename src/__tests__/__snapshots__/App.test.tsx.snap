// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
                "overflow": "hidden",
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="auto"
                  removeClippedSubviews={false}
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "opacity": 1,
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#eee",
                            "flex": 1,
                            "flexDirection": "column-reverse",
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            pointerEvents="auto"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 1,
                                "paddingTop": 64,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <View
                              accessibilityElementsHidden={false}
                              style={
                                Object {
                                  "backgroundColor": "#fff",
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                pointerEvents="auto"
                                style={
                                  Object {
                                    "alignItems": undefined,
                                    "backgroundColor": undefined,
                                    "borderRadius": undefined,
                                    "flex": 1,
                                    "flexDirection": undefined,
                                    "height": undefined,
                                    "justifyContent": undefined,
                                    "margin": undefined,
                                    "marginBottom": undefined,
                                    "marginLeft": undefined,
                                    "marginRight": undefined,
                                    "marginTop": undefined,
                                    "padding": undefined,
                                    "paddingBottom": undefined,
                                    "paddingHorizontal": undefined,
                                    "paddingLeft": undefined,
                                    "paddingRight": undefined,
                                    "paddingTop": undefined,
                                    "width": undefined,
                                  }
                                }
                              >
                                <View
                                  pointerEvents="auto"
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "#F5FCFF",
                                      "borderRadius": undefined,
                                      "flex": 1,
                                      "flexDirection": undefined,
                                      "height": undefined,
                                      "justifyContent": "center",
                                      "margin": undefined,
                                      "marginBottom": undefined,
                                      "marginLeft": undefined,
                                      "marginRight": undefined,
                                      "marginTop": undefined,
                                      "padding": undefined,
                                      "paddingBottom": undefined,
                                      "paddingHorizontal": undefined,
                                      "paddingLeft": undefined,
                                      "paddingRight": undefined,
                                      "paddingTop": undefined,
                                      "width": undefined,
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="auto"
                                    style={
                                      Object {
                                        "alignItems": undefined,
                                        "backgroundColor": undefined,
                                        "borderRadius": undefined,
                                        "flex": undefined,
                                        "flexDirection": undefined,
                                        "height": undefined,
                                        "justifyContent": undefined,
                                        "margin": undefined,
                                        "marginBottom": undefined,
                                        "marginLeft": undefined,
                                        "marginRight": undefined,
                                        "marginTop": undefined,
                                        "padding": undefined,
                                        "paddingBottom": 15,
                                        "paddingHorizontal": undefined,
                                        "paddingLeft": undefined,
                                        "paddingRight": undefined,
                                        "paddingTop": undefined,
                                        "width": undefined,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "color": "#333333",
                                          "fontSize": 24,
                                        }
                                      }
                                    >
                                      Welcome to Serto
                                      !
                                    </Text>
                                  </View>
                                </View>
                              </View>
                              <View
                                pointerEvents="auto"
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": undefined,
                                    "borderRadius": undefined,
                                    "bottom": 40,
                                    "flex": undefined,
                                    "flexDirection": undefined,
                                    "height": undefined,
                                    "justifyContent": undefined,
                                    "margin": undefined,
                                    "marginBottom": undefined,
                                    "marginLeft": undefined,
                                    "marginRight": undefined,
                                    "marginTop": undefined,
                                    "padding": undefined,
                                    "paddingBottom": undefined,
                                    "paddingHorizontal": undefined,
                                    "paddingLeft": undefined,
                                    "paddingRight": undefined,
                                    "paddingTop": undefined,
                                    "position": "absolute",
                                    "width": "100%",
                                  }
                                }
                              >
                                <View
                                  pointerEvents="auto"
                                  style={
                                    Object {
                                      "alignItems": undefined,
                                      "backgroundColor": undefined,
                                      "borderRadius": undefined,
                                      "flex": undefined,
                                      "flexDirection": undefined,
                                      "height": undefined,
                                      "justifyContent": undefined,
                                      "margin": undefined,
                                      "marginBottom": undefined,
                                      "marginLeft": undefined,
                                      "marginRight": undefined,
                                      "marginTop": undefined,
                                      "padding": undefined,
                                      "paddingBottom": undefined,
                                      "paddingHorizontal": undefined,
                                      "paddingLeft": undefined,
                                      "paddingRight": undefined,
                                      "paddingTop": undefined,
                                      "width": undefined,
                                    }
                                  }
                                >
                                  <View
                                    accessible={true}
                                    isTVSelectable={true}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#1EA17A",
                                        "borderRadius": 50,
                                        "height": 70,
                                        "justifyContent": "center",
                                        "opacity": 1,
                                        "width": 70,
                                      }
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#FFFFFF",
                                            "fontSize": 35,
                                          },
                                          undefined,
                                          Object {
                                            "fontFamily": "Ionicons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          Object {},
                                        ]
                                      }
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          pointerEvents="box-none"
                          style={
                            Object {
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "#FFF",
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <View
                              onLayout={[Function]}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "backgroundColor": "#FFF",
                                  "borderBottomColor": "#A7A7AA",
                                  "borderBottomWidth": 0.5,
                                  "height": 64,
                                  "paddingBottom": 0,
                                  "paddingLeft": 0,
                                  "paddingRight": 0,
                                  "paddingTop": 20,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "flex": 1,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "bottom": 0,
                                      "flexDirection": "row",
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    }
                                  }
                                >
                                  <View
                                    pointerEvents="box-none"
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "bottom": 0,
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "left": 70,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "right": 70,
                                        "top": 0,
                                      }
                                    }
                                  >
                                    <Text
                                      accessibilityTraits="header"
                                      allowFontScaling={false}
                                      numberOfLines={1}
                                      onLayout={[Function]}
                                      style={
                                        Object {
                                          "color": "rgba(0, 0, 0, .9)",
                                          "fontSize": 17,
                                          "fontWeight": "600",
                                          "marginHorizontal": 16,
                                          "textAlign": "center",
                                        }
                                      }
                                    >
                                      Serto
                                    </Text>
                                  </View>
                                  <View
                                    pointerEvents="box-none"
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "bottom": 0,
                                        "flexDirection": "row",
                                        "left": 0,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "top": 0,
                                      }
                                    }
                                  >
                                    <View
                                      pointerEvents="auto"
                                      style={
                                        Object {
                                          "alignItems": undefined,
                                          "backgroundColor": undefined,
                                          "borderRadius": undefined,
                                          "flex": undefined,
                                          "flexDirection": undefined,
                                          "height": undefined,
                                          "justifyContent": undefined,
                                          "margin": undefined,
                                          "marginBottom": undefined,
                                          "marginLeft": undefined,
                                          "marginRight": undefined,
                                          "marginTop": undefined,
                                          "padding": undefined,
                                          "paddingBottom": undefined,
                                          "paddingHorizontal": undefined,
                                          "paddingLeft": 15,
                                          "paddingRight": undefined,
                                          "paddingTop": undefined,
                                          "width": undefined,
                                        }
                                      }
                                    >
                                      <View
                                        accessible={true}
                                        isTVSelectable={true}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderRadius": 8,
                                            "maxWidth": 300,
                                          }
                                        }
                                      >
                                        <View
                                          pointerEvents="auto"
                                          style={
                                            Object {
                                              "alignItems": undefined,
                                              "backgroundColor": undefined,
                                              "borderRadius": undefined,
                                              "flex": undefined,
                                              "flexDirection": "row",
                                              "height": undefined,
                                              "justifyContent": "center",
                                              "margin": undefined,
                                              "marginBottom": undefined,
                                              "marginLeft": undefined,
                                              "marginRight": undefined,
                                              "marginTop": undefined,
                                              "padding": undefined,
                                              "paddingBottom": undefined,
                                              "paddingHorizontal": undefined,
                                              "paddingLeft": undefined,
                                              "paddingRight": undefined,
                                              "paddingTop": undefined,
                                              "width": undefined,
                                            }
                                          }
                                        >
                                          <View
                                            pointerEvents="auto"
                                            style={
                                              Object {
                                                "alignItems": undefined,
                                                "backgroundColor": undefined,
                                                "borderRadius": undefined,
                                                "flex": undefined,
                                                "flexDirection": undefined,
                                                "height": undefined,
                                                "justifyContent": undefined,
                                                "margin": undefined,
                                                "marginBottom": undefined,
                                                "marginLeft": undefined,
                                                "marginRight": 10,
                                                "marginTop": undefined,
                                                "padding": undefined,
                                                "paddingBottom": undefined,
                                                "paddingHorizontal": undefined,
                                                "paddingLeft": undefined,
                                                "paddingRight": undefined,
                                                "paddingTop": undefined,
                                                "width": undefined,
                                              }
                                            }
                                          >
                                            <Text
                                              allowFontScaling={false}
                                              style={
                                                Array [
                                                  Object {
                                                    "color": "#333333",
                                                    "fontSize": 32,
                                                  },
                                                  undefined,
                                                  Object {
                                                    "fontFamily": "Ionicons",
                                                    "fontStyle": "normal",
                                                    "fontWeight": "normal",
                                                  },
                                                  Object {},
                                                ]
                                              }
                                            >
                                              
                                            </Text>
                                          </View>
                                          <View
                                            pointerEvents="auto"
                                            style={
                                              Object {
                                                "alignItems": "center",
                                                "backgroundColor": undefined,
                                                "borderRadius": undefined,
                                                "flex": undefined,
                                                "flexDirection": undefined,
                                                "height": undefined,
                                                "justifyContent": "center",
                                                "margin": undefined,
                                                "marginBottom": undefined,
                                                "marginLeft": undefined,
                                                "marginRight": undefined,
                                                "marginTop": undefined,
                                                "padding": undefined,
                                                "paddingBottom": undefined,
                                                "paddingHorizontal": undefined,
                                                "paddingLeft": undefined,
                                                "paddingRight": undefined,
                                                "paddingTop": undefined,
                                                "width": undefined,
                                              }
                                            }
                                          >
                                            <Text
                                              style={
                                                Object {
                                                  "fontSize": 18,
                                                }
                                              }
                                            />
                                          </View>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "black",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 1000,
                  }
                }
              />
            </View>
            <View
              accessibilityViewIsModal={false}
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": -320,
                    },
                  ],
                  "zIndex": 1001,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "width": 320,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <RCTScrollView
                    alwaysBounceVertical={false}
                  >
                    <View>
                      <View
                        onLayout={[Function]}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "paddingBottom": 0,
                            "paddingLeft": 0,
                            "paddingRight": 0,
                            "paddingTop": 20,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "paddingVertical": 4,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessibilityLabel="Home"
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              onLayout={[Function]}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "rgba(0, 0, 0, .04)",
                                  "flexDirection": "row",
                                  "paddingBottom": 0,
                                  "paddingLeft": 0,
                                  "paddingRight": 0,
                                  "paddingTop": 0,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "fontWeight": "bold",
                                      "margin": 16,
                                    },
                                    Object {
                                      "color": "#2196f3",
                                    },
                                    undefined,
                                    undefined,
                                  ]
                                }
                              >
                                Home
                              </Text>
                            </View>
                          </View>
                          <View
                            accessibilityLabel="Developer"
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              onLayout={[Function]}
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "flexDirection": "row",
                                  "paddingBottom": 0,
                                  "paddingLeft": 0,
                                  "paddingRight": 0,
                                  "paddingTop": 0,
                                }
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "fontWeight": "bold",
                                      "margin": 16,
                                    },
                                    Object {
                                      "color": "rgba(0, 0, 0, .87)",
                                    },
                                    undefined,
                                    undefined,
                                  ]
                                }
                              >
                                Developer
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
